fastly/client.go

```diff
@@ -198,7 +203,13 @@ func (c *Client) Request(verb, p string, ro *RequestOptions) (*http.Response, er
 		return nil, err
 	}
 
+	if verb != "GET" && verb != "HEAD" {
+		c.updateLock.Lock()
+	}
 	resp, err := checkResp(c.HTTPClient.Do(req))
+	if verb != "GET" && verb != "HEAD" {
+		c.updateLock.Unlock()
+	}
```

Comment:
We can avoid the need for this additional block and risk the conditions getting out of sync by using a `defer` statement directly after your `.Lock()` above. Such as:

```
if verb != "GET" && verb != "HEAD" {
		c.updateLock.Lock()
		defer c.updateLock.Unlock()
}
```